### DISCUSSION TITLE
Elasticsearch not generating token

### DISCUSSION THREAD

**Discussion by Quarrymann** [pfelk/pfelk#504] [General]
It never fails. I am a noob for linux, but I have spent many hours on these auto installs and never have I had one complete w/o errors.
I ran 14 install last year using docker compose and every one failed for different errors.

I finally see there is a new script version....awesome.
Installed Debian 12--- ran the scripts.... loosk good. Get to the end and no token generated. SSL cert error. uuhg.
Gonna try again on ubuntu 22.04- will update tomorrow.

**Comment by a3ilson** [comment]
@Quarrymann did you have any luck? I’d be curious as to what’s causing the issue. However, docker hasn’t been my go-to as the easiest to setup and get up and running.

**Reply by Quarrymann** [reply]
I did not. I used the scripted install this month and it fails to issue token for kibana. I might try docker ...or maybe the solo installs. Bit defeated as I've played with this on and off for a year+. Will have to look.

**Reply by a3ilson** [reply]
Sand and sorry to hear about your troubles.  I could create a vm preinstalled.

however, I’d be more interested in troubleshooting. What is the system specs in which you are installing to?

Are you able to send the logs from elastic (search, kibana)?

With your latest/current attempt…did elastic, kibana and logstash install?

**Reply by a3ilson** [reply]
Let me know what timezone you’re in and if you’d like to troubleshoot via discord etc?

**Reply by Quarrymann** [reply]
Pacific--

**Comment by Quarrymann** [comment]
I am running Ubuntu 22.04.3.
Its an HP z3 g3 16GB on m.2 drive 500GB
i7-7700 3.6gHz
Should be stout enough to at least install. I am on the step by step  install right now.
https://github.com/pfelk/pfelk/blob/main/install/ubuntu.md 
Step 9

"The exact error was "must set  xpack.security.enrollment.enabled true" 
The best research I did said this belongs in elacticsearch.yml. 

Also, thank you for replying. I was not ever expecting help on my rant.

James

**Reply by a3ilson** [reply]
You received that error during step 9? Or from before?

everything should be smooth sailing…the security settings can trip you up though.

I’ll stick around for a bit (hour or so). Keep me posted.

**Reply by Quarrymann** [reply]
Wow-- thanks. 
That was from last week. Sunday.
I am running into duplicate files from directies that remained...so cleaning them up before d/l again.
At step 17-- so far so good.

**Comment by Quarrymann** [comment]
And step 22 wont run.

sed -i 's?ExecStart=/usr/share/logstash/bin/logstash "--path.settings" "/etc/logstash"?ExecStart=/usr/share/logstash/bin/logstash "--pipeline.unsafe_shutdown" "--path.settings" "/etc/logstash"?' /etc/systemd/system/logstash.service

sed: can't read /etc/systemd/system/logstash.service: No such file or directory

**Reply by a3ilson** [reply]
Skip 22…
I’m make note and revise, if/as needed. It isn’t required.

**Reply by Quarrymann** [reply]
ok. everything started up.
Clciked over to next tab-- 0️⃣ Security

This is were it failed before-> sudo /usr/share/elasticsearch/bin/elasticsearch-create-enrollment-token --scope kibana

And failed just now
ERROR: [xpack.security.enrollment.enabled] must be set to `true` to create an enrollment token

**Reply by Quarrymann** [reply]
I can go to discord channel if its faster communication

**Reply by a3ilson** [reply]
created a server...

pfelk 

https://discord.gg/pYqfhUvW

**Comment by a3ilson** [comment]
run this, `sudo cat /etc/elasticsearch/elasticsearch.yml`

It should display the Elasticsearch.yml file...let me know if it looks like this or a # precedes the setting:

```
# Enable security features
xpack.security.enabled: true
```

**Comment by Quarrymann** [comment]
Gandorf

**Comment by Quarrymann** [comment]
Elasticsearch is not starting now.

elasticsearch.service - Elasticsearch
     Loaded: loaded (/lib/systemd/system/elasticsearch.service; enabled;
vendor>
     Active: failed (Result: exit-code) since Fri 2023-09-01 20:33:21 PDT;
1min>
       Docs: https://www.elastic.co
    Process: 6148 ExecStart=/usr/share/elasticsearch/bin/systemd-entrypoint
-p >
   Main PID: 6148 (code=exited, status=1/FAILURE)
        CPU: 46.186s

Sep 01 20:33:05 elk-server systemd[1]: Starting Elasticsearch...
Sep 01 20:33:20 elk-server systemd-entrypoint[6148]: ERROR: Elasticsearch
did n>
Sep 01 20:33:21 elk-server systemd-entrypoint[6148]: ERROR: Elasticsearch
exite>
Sep 01 20:33:21 elk-server systemd[1]: elasticsearch.service: Main process
exit>
Sep 01 20:33:21 elk-server systemd[1]: elasticsearch.service: Failed with
resul>
Sep 01 20:33:21 elk-server systemd[1]: Failed to start Elasticsearch.
Sep 01 20:33:21 elk-server systemd[1]: elasticsearch.service: Consumed
46.186s >

-----------------------------
journalctl -xeu elasticsearch.service
░░ An ExecStart= process belonging to unit elasticsearch.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 1.
Sep 01 20:35:25 elk-server systemd[1]: elasticsearch.service: Failed with
resul>
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ The unit elasticsearch.service has entered the 'failed' state with
result 'e>
Sep 01 20:35:25 elk-server systemd[1]: Failed to start Elasticsearch.
░░ Subject: A start job for unit elasticsearch.service has failed
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ A start job for unit elasticsearch.service has finished with a failure.
░░
░░ The job identifier is 8114 and the job result is failed.
Sep 01 20:35:25 elk-server systemd[1]: elasticsearch.service: Consumed
46.025s >
░░ Subject: Resources consumed by unit runtime
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ The unit elasticsearch.service completed and consumed the indicated
resource>

**Comment by Quarrymann** [comment]
If I comment out all the lines you added then ES will run.

$ systemctl status elasticsearch
● elasticsearch.service - Elasticsearch
     Loaded: loaded (/lib/systemd/system/elasticsearch.service; enabled;
vendor>
     Active: active (running) since Fri 2023-09-01 20:42:29 PDT; 11s ago
       Docs: https://www.elastic.co
   Main PID: 9774 (java)
      Tasks: 120 (limit: 18954)
     Memory: 8.5G
        CPU: 1min 12.596s
     CGroup: /system.slice/elasticsearch.service
             ├─9774 /usr/share/elasticsearch/jdk/bin/java -Xms4m -Xmx64m
-XX:+U>
             ├─9837 /usr/share/elasticsearch/jdk/bin/java
-Des.networkaddress.c>
             └─9865
/usr/share/elasticsearch/modules/x-pack-ml/platform/linux-x>

Sep 01 20:42:09 elk-server systemd[1]: Starting Elasticsearch...
Sep 01 20:42:29 elk-server systemd[1]: Started Elasticsearch.

**Comment by Quarrymann** [comment]
This is normal for my experience.

failed something-- how is it this installs everything except the last
part---- ????? i can sudo nano that file but it is empty. but it does
exist. And it was created by the apt install---
Is there some base program I am missing to pass sudo permissions.... it is
just so frustrating.


Selecting previously unselected package logstash.
(Reading database ... 207871 files and directories currently installed.)
Preparing to unpack .../logstash_1%3a8.9.1-1_amd64.deb ...
Unpacking logstash (1:8.9.1-1) ...
Setting up logstash (1:8.9.1-1) ...
sed: can't read /etc/logstash/logstash.yml: No such file or directory
chmod: cannot access '/etc/logstash/startup.options': No such file or
directory
Scanning processes...

**Comment by Quarrymann** [comment]
*I went through every line of code your screen shows during install and
they differ as well.*
*Elasticsearch installs first and I notice you have these 2 lines that I do
not.*
*->creating elasticsearch group...ok*
*->creating elasticsearch user...ok*

I do not see those lines during my install. It skips them

copy/paste from my install
----------------
Setting up elasticsearch (8.9.1) ...
Created elasticsearch keystore in /etc/elasticsearch/elasticsearch.keystore
Scanning processes...

Scanning processor microcode...

Scanning linux images...


Running kernel seems to be up-to-date.
-----------------------------------


*Same thing in Kibana--- missing same 2 lines*
*->creating kibana group... ok*
*->creating kibana user... ok*
-----------------------
The following NEW packages will be installed:
  kibana
0 upgraded, 1 newly installed, 0 to remove and 1 not upgraded.
Need to get 288 MB of archives.
After this operation, 768 MB of additional disk space will be used.
Get:1 https://artifacts.elastic.co/packages/8.x/apt stable/main amd64
kibana amd64 8.9.1 [288 MB]
Fetched 288 MB in 7s (42.8 MB/s)

Selecting previously unselected package kibana.
(Reading database ... 130695 files and directories currently installed.)
Preparing to unpack .../kibana_8.9.1_amd64.deb ...
Unpacking kibana (8.9.1) ...
Setting up kibana (8.9.1) ...
Restarting kibana service... OK
Created Kibana keystore in /etc/kibana/kibana.keystore
Scanning processes...

Scanning processor microcode...

Scanning linux images...


Running kernel seems to be up-to-date.
-------------------
[image: image.png]

**Comment by a3ilson** [comment]
The group & users were likely created from a prior instance. I do not believe they are removed when removing Elasticsearch/Kibana/Logstash.

I doubt that'll be any issue... however, when you removed via `sudo apt-get remove elasticsearch kibana logstash` 

I'd give this a try instead, which should remove all files system-wide by purging all data `sudo apt-get purge elasticsearch kibana logstash`

**Comment by Quarrymann** [comment]
Did not remove everything

sudo apt-get purge elasticsearch kibana logstash
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following packages will be REMOVED:
  elasticsearch* kibana* logstash*
0 upgraded, 0 newly installed, 3 to remove and 1 not upgraded.
After this operation, 2,619 MB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 222485 files and directories currently installed.)
Removing elasticsearch (8.9.1) ...
Stopping elasticsearch service... OK
Removing kibana (8.9.1) ...
Stopping kibana service... OK
Removing logstash (1:8.9.1-1) ...
(Reading database ... 129397 files and directories currently installed.)
Purging configuration files for elasticsearch (8.9.1) ...
Deleting security auto-configuration directory...OK
Deleting elasticsearch.keystore...OK
dpkg: warning: while removing elasticsearch, directory
'/var/lib/elasticsearch' not empty so not removed
Purging configuration files for kibana (8.9.1) ...
dpkg: warning: while removing kibana, directory '/etc/kibana' not empty so
not removed
dpkg: warning: while removing kibana, directory '/var/lib/kibana' not empty
so not removed
dpkg: warning: while removing kibana, directory '/usr/lib/systemd/system'
not empty so not removed
Purging configuration files for logstash (1:8.9.1-1) ...
dpkg: warning: while removing logstash, directory '/var/lib/logstash' not
empty so not removed
dpkg: warning: while removing logstash, directory '/var/log/logstash' not
empty so not removed

**Reply by a3ilson** [reply]
Short of just reinstalling the OS.  I know the feeling and still perplexed as to what the culprit might be. 

Fresh install OS and try again? Or give docker a try? I can also build a preconfigured VM and send it.
